src/modules/job-list/job_data.c

```diff
@@ -118,6 +118,71 @@ static const char *parse_job_name (const char *path)
     return path;
 }
 
+static int parse_jobspec_job_name (struct job *job,
+                                   json_t *jobspec_job,
+                                   json_t *tasks)
+{
+    json_error_t error;
+    json_t *command = NULL;
+
+    if (json_unpack_ex (tasks, &error, 0,
+                        "[{s:o}]",
+                        "command", &command) < 0) {
+        flux_log (job->h, LOG_ERR,
+                  "%s: job %ju invalid jobspec: %s",
+                  __FUNCTION__, (uintmax_t)job->id, error.text);
+        return -1;
+    }
+
+    if (!json_is_array (command)) {
+        flux_log (job->h, LOG_ERR,
+                  "%s: job %ju invalid jobspec",
+                  __FUNCTION__, (uintmax_t)job->id);
+        return -1;
+    }
```

Comment:
Now that this code is in its own function, it is clear this block of code is unused when an explicit job name is set in jobspec. This stuff could be moved into the `if (!job->name)` block and therefore only executed when the job does not otherwise already have a name set.